mantidproject/mantid
# Fix matrix and table workspace display widgets not staying ontop

When ontop window behaviour was set, the matrix and table workspace display widgets would continue to act as floating. I didn't see this on windows. This PR modifies which widget recieves the window flags so that the widgets have the correct on top behaviour.



**To test:**
1. Ensure ontop windows are set in `Settings > General`
2. Load a matrix workspace
3. Right click on it, `Show Data`
4. Click back onto the main workbench window
5. Ensure the workspace data display stays on top.
6. Change the window behaviour to floating in `Settings > General`
7. Reopen the display and ensure that the ontop behaviour no longer occurs
6. Repeat for a table workspace (you can do this with the `CreateDetectorTable` algorithm)


Fixes #31305 . 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
